test/kb_Metrics_server_test.py

```diff
@@ -118,6 +205,596 @@ def getImpl(self):
+        self.assertFalse(ws_narrs[23]['deleted'])
+        self.assertEqual(ws_narrs[23]['desc'], '')
+        self.assertEqual(ws_narrs[23]['numObj'], 4)
+        self.assertEqual(ws_narrs[23]['last_saved_at'],
+                         datetime.datetime(2018, 1, 24, 19, 35, 30, 1000))
+
+        # testing the broadest time range in the db
+        earliest = 1468454614192
+        latest = 1516822530001
+        ws_narrs = self.dbi.list_ws_narratives(earliest, latest)
+        self.assertEqual(len(ws_narrs), 24)
+
+        # testing only given the lower bound
+        ws_narrs = self.dbi.list_ws_narratives(minT=earliest)
+        self.assertEqual(len(ws_narrs), 24)
+
+        # testing only given the upper bound
+        ws_narrs = self.dbi.list_ws_narratives(maxT=latest)
+        self.assertEqual(len(ws_narrs), 24)
+
+        # testing swap the lower/upper bounds
+        ws_narrs = self.dbi.list_ws_narratives(minT=latest, maxT=earliest)
+        self.assertEqual(len(ws_narrs), 24)
+
+    # Uncomment to skip this test
+    # @unittest.skip("skipped test_MetricsMongoDBs_list_ujs_results")
+    def test_MetricsMongoDBs_list_ujs_results(self):
+        minTime = 1500000932952
+        maxTime = 1500046845591
+        user_list1 = ['tgu2', 'umaganapathyswork', 'arfath']
+        user_list2 = ['umaganapathyswork', 'arfath']
+
+        epoch = datetime.datetime.utcfromtimestamp(0)
+        # testing list_ujs_results return data, with userIds
+        ujs = self.dbi.list_ujs_results(user_list1, minTime, maxTime)
+        self.assertEqual(len(ujs), 14)
+        ujs = self.dbi.list_ujs_results(user_list2, minTime, maxTime)
+        self.assertEqual(len(ujs), 3)
+        for uj in ujs:
+            self.assertIn(uj.get('user'), user_list2)
+            uj_creation_time = int((uj.get('created') -
+                                    epoch).total_seconds() * 1000.0)
+            self.assertTrue(minTime <= uj_creation_time <= maxTime)
+
+        # testing list_ujs_results return data, without userIds
+        ujs = self.dbi.list_ujs_results([], minTime, maxTime)
+        self.assertEqual(len(ujs), 14)
+
+        # testing list_ujs_results return data, different userIds and times
+        ujs = self.dbi.list_ujs_results(['wjriehl'],
+                                        1500052541065, 1500074641912)
+        self.assertEqual(len(ujs), 8)
+        ujs = self.dbi.list_ujs_results([], 1500052541065, 1500074641912)
+        self.assertEqual(len(ujs), 14)
+
+    # Uncomment to skip this test
+    # @unittest.skip("skipped test_MetricsMongoDBController_constructor")
+    def test_MetricsMongoDBController_constructor(self):
+        cfg_arr = self.cfg
+        # testing if all the required parameters are given
+        with self.assertRaises(ValueError):
+            try:
+                cfg_arr = self.cfg
+                db_ctr1 = MetricsMongoDBController(cfg_arr)
+            except ValueError:
```

Comment:
I am not sure why you want to do except: pass in a test.

Reply:
Here is to test if the required parameters are given, no exception is raised.

Reply:
I see. Then you are not testing correctly. If you want to test no raise, you can just do 
`db_ctr1 = MetricsMongoDBController(cfg_arr)` then see db_ctr1 has expected data.

Your logic here is try the func, if the func finishes then raise ValueError and then catch it. I don't see any benefit to make it this complicated.

Reply:
okay